substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/configure/ConfigurationParser.java

```diff
@@ -74,9 +74,39 @@ public void parseAndRegister(Path path) throws IOException {
         throw new JSONParserException(errorMessage);
     }
 
+    protected void checkExclusiveOptionsAttributes(Map<String, Object> map, String type, Collection<String> requiredAttrs, Collection<String> optionalAttrs) {
+        Set<String> unseenRequired = new HashSet<>(requiredAttrs);
+        for (String attribute : map.keySet()) {
+            if (unseenRequired.contains(attribute)) {
+                unseenRequired.removeAll(requiredAttrs);
+                break;
+            }
+        }
+
+        checkRemainingAttributes(map, type, requiredAttrs, optionalAttrs, unseenRequired);
+    }
+
     protected void checkAttributes(Map<String, Object> map, String type, Collection<String> requiredAttrs, Collection<String> optionalAttrs) {
         Set<String> unseenRequired = new HashSet<>(requiredAttrs);
         unseenRequired.removeAll(map.keySet());
+        checkRemainingAttributes(map, type, requiredAttrs, optionalAttrs, unseenRequired);
+    }
+
+    protected void warnOrFail(String message) {
```

Comment:
No need to move these functions, it will just mess up the history.

Reply:
Order of the functions is not changed now (comparing to master branch).